cns/service/main.go

```diff
@@ -1303,6 +1303,52 @@ func InitializeCRDState(ctx context.Context, httpRestService cns.HTTPService, cn
 	// TODO(rbtr): nodename and namespace should be in the cns config
 	directscopedcli := nncctrl.NewScopedClient(directnnccli, types.NamespacedName{Namespace: "kube-system", Name: nodeName})
 
+	// Create the base NNC CRD if HomeAz is enabled
+	if cnsconfig.EnableHomeAz {
+		homeAzResponse := httpRestServiceImplementation.GetHomeAz(ctx)
+		az := homeAzResponse.HomeAzResponse.HomeAz
+		logger.Printf("[Azure CNS] HomeAz: %d", az)
+		// Create Node Network Config CRD and update the Home Az field with the cache value from the HomeAz Monitor
+		var nnc *v1alpha.NodeNetworkConfig
+		if nnc, err = directnnccli.Get(ctx, types.NamespacedName{Namespace: "kube-system", Name: nodeName}); err != nil {
+			logger.Errorf("[Azure CNS] failed to get existing NNC: %v", err)
```

Comment:
There's no return here, so this will continue. Is that intentional?

Reply:
Yes, that is intentional as we could have a scenario where the NNC does not exist while we are trying to move the NNC creation logic gradually over to CNS.

Reply:
Might be worth leaving a comment to that effect.

Reply:
Might take up Ramiro's suggestion and add a retrier incase of network error in reaching the API server or atleast checking if the error is clearly a NNC Not Found error before proceeding

Reply:
Logic here has changed.